Mlem/API/Internal/HierarchicalComment.swift

```diff
@@ -22,11 +22,14 @@ class HierarchicalComment: ObservableObject {
     @Published var isCollapsed: Bool = false
 
     init(comment: APICommentView, children: [HierarchicalComment], parentCollapsed: Bool, collapsed: Bool) {
+        let collapseChildCommentsFlag = UserDefaults.standard.bool(forKey: "collapseChildComments")
+        let depth = max(0, comment.comment.path.split(separator: ".").count - 2)
+        
         self.commentView = comment
         self.children = children
-        self.depth = max(0, commentView.comment.path.split(separator: ".").count - 2)
-        self.isParentCollapsed = parentCollapsed
-        self.isCollapsed = collapsed
+        self.depth = depth
+        self.isParentCollapsed = collapseChildCommentsFlag && depth > 1 || parentCollapsed
+        self.isCollapsed = collapseChildCommentsFlag && depth == 1 || collapsed
```

Comment:
What do you think about collapsing all comments, not just the top layer? That way, when you expand a reply, that reply's replies are also collapsed until tapped.

```suggestion
        self.isCollapsed = collapseChildCommentsFlag && depth >= 1 || collapsed
```

Reply:
I was considering that but this is closer to how Apollo had it where all the comments would load after you uncollapse one. In the future I'll submit something closer, where it just shows "x amount of replies" under the parent comment and it uncollapses all of the children when you click that. 


Up to you though, I can make it collapse everything if you want.

Reply:
I agree with the way it currently is (expanding one reply expands its whole sub-thread)--that feels like a more natural flow

Reply:
Sure, that's fine 👍 

I agree that in the future an "expand n comments" button, reminiscent of what Apollo had, would be a better experience.